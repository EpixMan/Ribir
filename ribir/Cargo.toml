[package]
authors.workspace = true
categories.workspace = true
description.workspace = true
documentation.workspace = true
edition.workspace = true
homepage.workspace = true
keywords.workspace = true
license.workspace = true
name = "ribir"
readme.workspace = true
repository = "https://github.com/RibirX/Ribir/ribir"
version.workspace = true

[dependencies]
ribir_core = {path = "../core", version = "0.0.0"}
ribir_gpu = {path = "../gpu", version = "0.0.0"}
ribir_widgets = {path = "../widgets", version = "0.0.0"}
winit.workspace = true

[dev-dependencies]
env_logger.workspace = true
image.workspace = true
ribir_material = {path = "../themes/material", version = "0.0.0"}

[features]
default = ["wgpu"]
png = ["ribir_core/png"]
wgpu = ["ribir_gpu/wgpu"]
