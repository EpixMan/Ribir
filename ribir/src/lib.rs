pub use ribir_core as core;
pub use ribir_widgets as widgets;

pub mod app;
mod winit_shell_wnd;

pub mod prelude {
  pub use crate::app::*;
  pub use ribir_core::prelude::*;
  pub use ribir_widgets::prelude::*;
}
