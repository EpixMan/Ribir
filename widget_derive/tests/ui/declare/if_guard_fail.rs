use ribir::prelude::*;

fn normal_if_guard_pass(ctx: &mut BuildCtx) {
  let guard = Some(1);
  declare! {
    SizedBox {
      // if guard in widget's field
      size if true => : Size::zero(),
      // if guard in data atribute
      cursor if true => : CursorIcon::Hand,
      // if guard in listener attibute
      on_tap if let Some(_) = guard  => : |_| {},
      // if guard in wrap widget.
      margin if true => : EdgeInsets::all(1.)
    }
  };
}

fn id_if_guard_fail(ctx: &mut BuildCtx) {
  let guard = Some(1);
  declare! {
    SizedBox {
      id if true => : test,
      // if guard in widget's field
      size if true => : Size::zero(),

    }
  };
}

fn depend_id_behind_if_guard_fail(ctx: &mut BuildCtx) {
  declare! {
    SizedBox {
      id: a,
      size: Size::zero(),
      margin if true =>:  EdgeInsets::all(0.),

      SizedBox{
        size: Size::zero(),
        margin: a.margin
      }

    }
  };
}

fn main() {}
