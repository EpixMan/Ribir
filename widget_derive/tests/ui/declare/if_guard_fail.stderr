error: if guard is not supported in `id`
  --> tests/ui/declare/if_guard_fail.rs:23:10
   |
23 |       id if true => : test,
   |          ^^^^^^^

error: Depends on a widget field which behind `if guard`, its existence depends on the `if guard` result in runtime.
  --> tests/ui/declare/if_guard_fail.rs:40:17
   |
40 |         margin: a.margin
   |                 ^^^^^^^^
   |
warning: field define here.
  --> tests/ui/declare/if_guard_fail.rs:34:11
   |
34 |       id: a,
   |           ^
35 |       size: Size::zero(),
36 |       margin if true =>:  EdgeInsets::all(0.),
   |       ^^^^^^ ^^^^^^^
